matchms/filtering/require_annotation.py

```diff
@@ -0,0 +1,39 @@
+import logging
+from matchms import Spectrum
+from matchms.metadata_utils import (is_valid_inchi, is_valid_inchikey,
+                                    is_valid_smiles, convert_smiles_to_inchi,
+                                    convert_inchi_to_inchikey)
+
+logger = logging.getLogger("matchms")
+
+
+def require_annotation(spectrum_in: Spectrum):
+    """Removes spectra that are not fully annotated (correct and matching, smiles, inchi and inchikey)"""
+    if spectrum_in is None:
+        return None
+    spectrum = spectrum_in.clone()
+    smiles = spectrum.get("smiles")
+    inchi = spectrum.get("inchi")
+    inchikey = spectrum.get("inchikey")
```

Comment:
I think we don't need to clone here as the spectrum variable is never used except when returning.